### ISSUE TITLE
TS Server very slow - "Code Helper" process uses 100%+ CPU

### ISSUE THREAD

**Issue by bwanicur** [microsoft/vscode#83110]
- VSCode Version:
1.39.2

- OS Version:
MacOS 14.6

Steps to Reproduce:

1.  Run VsCode without extensions
2.  Open Project

Does this issue occur when all extensions are disabled?: Yes/No

Yes.  After a few seconds the "Code Helper" subprocess will start to consume 100%+ CPU power.

```
Version:          Code 1.39.2 (6ab598523be7a800d7f3eb4d92d7ab9a66069390, 2019-10-15T15:33:00.827Z)
OS Version:       Darwin x64 18.7.0
CPUs:             Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)
Memory (System):  16.00GB (1.99GB free)
Load (avg):       3, 3, 2
VM:               0%
Screen Reader:    no
Process Argv:     --disable-extensions .
GPU Status:       2d_canvas:                     enabled
                  flash_3d:                      enabled
                  flash_stage3d:                 enabled
                  flash_stage3d_baseline:        enabled
                  gpu_compositing:               enabled
                  multiple_raster_threads:       enabled_on
                  native_gpu_memory_buffers:     enabled
                  oop_rasterization:             disabled_off
                  protected_video_decode:        unavailable_off
                  rasterization:                 enabled
                  skia_deferred_display_list:    disabled_off
                  skia_renderer:                 disabled_off
                  surface_synchronization:       enabled_on
                  video_decode:                  enabled
                  viz_display_compositor:        disabled_off
                  webgl:                         enabled
                  webgl2:                        enabled

CPU %	Mem MB	   PID	Process
    0	    82	 43226	code main
    0	    49	 43265	   gpu-process
    0	   262	 43281	   window (FilterState.js — db-saas)
    0	    66	 43284	     extensionHost
    0	    82	 43390	       electron_node tsserver.js
  202	  2146	 43391	       electron_node tsserver.js
    0	    82	 43392	         electron_node typingsInstaller.js typesMap.js
    0	    49	 43285	     watcherService
    0	    82	 43305	   shared-process
    0	     0	 43462	     /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=

Workspace Stats:
|  Window (FilterState.js — db-saas)
|    Folder (db-saas): 1757 files
|      File types: js(523) jsx(327) scss(219) png(110) json(78) svg(67)
|                  html(50) md(8) jpg(7) csv(5)
|      Conf files: package.json(6) settings.json(1)
```

**Comment by mjbvz** [comment]
* Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

* Do you see any errors or warning printed in the developer console? You can view this using `Developer: Toggle developer tools`

* Also please try upgrading your workspace to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) to see if that fixes the issue

**Comment by mjbvz** [comment]
Also spikes of cpu usage are expected when you first open large projects. Does the high cpu usage stop?

**Comment by mjbvz** [comment]
But the cpu usage go away after a moment or two?

**Comment by bwanicur** [comment]
No.  At first it appears to be intermittent, but if I leave VSCode running, the process will eventually become constant and the fan will engage.  The fan noise is how I originally found this issue :0 .

**Comment by mjbvz** [comment]
Does this only happen when you open a specific project? Also, where there any errors in the developer console? You can check this using the `Developer: Toggle developer tools` command

**Comment by bwanicur** [comment]
No errors in the console.  Sorry I forgot to include that.  I did find the answer to my issue!  And this was only happening when I opened projects with a relatively large number of files (node_modules).

I was excluding the node_modules/ files in the "User" settings but not my "workspace" settings.  I was very confused by this as I thought that my user settings would apply to the workspace ?   Once I excluded the node_modules/ directory in my workspace settings, the CPU spikes ceased.  Apologies for the wild goose chase.
